lib/rust/ensogl/component/grid-view/src/basic.rs

```diff
@@ -0,0 +1,182 @@
+    pub const PADDING_PX: f32 = 5.0;
+
+    ensogl_core::define_shape_system! {
+        (style:Style, color: Vector4) {
+            let shape_width  : Var<Pixels> = "input_size.x".into();
+            let shape_height : Var<Pixels> = "input_size.y".into();
+            let width = shape_width - 2.0.px() * PADDING_PX;
+            let height = shape_height - 2.0.px() * PADDING_PX;
+            Rect((width, height)).fill(color).into()
+        }
+    }
+}
+
+
+
+// ===================
+// === EntryParams ===
+// ===================
+
+/// The parameters of [`BasicGridView`]`s entries.
+#[allow(missing_docs)]
+#[derive(Clone, Debug)]
+pub struct EntryParams {
+    pub bg_color:    color::Rgba,
+    pub bg_margin:   f32,
+    pub font:        ImString,
+    pub text_offset: f32,
+    pub text_size:   text::Size,
+    pub text_color:  color::Rgba,
+}
+
+impl Default for EntryParams {
+    fn default() -> Self {
+        Self {
+            bg_color:    color::Rgba::transparent(),
+            bg_margin:   0.0,
+            font:        text::typeface::font::DEFAULT_FONT.into(),
+            text_offset: 7.0,
+            text_size:   text::Size(14.0),
+            text_color:  default(),
+        }
+    }
+}
+
+
+
+// =============
+// === Entry ===
+// =============
+
+// === EntryData ===
+
+/// An internal structure of [`Entry`], which may be passed to FRP network without risk of memory
+/// leaks.
+#[allow(missing_docs)]
+#[derive(Clone, Debug)]
+pub struct EntryData {
+    display_object: display::object::Instance,
+    pub label:      text::Area,
+    pub background: entry_background::View,
+}
+
+impl EntryData {
+    fn new(app: &Application, text_layer: &Option<Layer>) -> Self {
+        let logger = Logger::new("list_view::entry::Label");
+        let display_object = display::object::Instance::new(&logger);
+        let label = app.new_view::<ensogl_text::Area>();
+        let background = entry_background::View::new(&logger);
+        display_object.add_child(&label);
+        display_object.add_child(&background);
+        if let Some(layer) = text_layer {
+            label.add_to_scene_layer(layer);
+        }
+        Self { display_object, label, background }
+    }
+
+    fn update_layout(
+        &self,
+        entry_size: Vector2,
+        bg_margin: f32,
+        text_size: text::Size,
+        text_offset: f32,
+    ) {
+        use entry_background::PADDING_PX;
+        let bg_size = entry_size - Vector2(bg_margin, bg_margin) * 2.0;
+        let bg_size_with_padding = bg_size + Vector2(PADDING_PX, PADDING_PX) * 2.0;
+        self.background.size.set(bg_size_with_padding);
+        self.label.set_position_xy(Vector2(text_offset - entry_size.x / 2.0, text_size.raw / 2.0));
+    }
+}
+
+
+// === Entry ===
+
+/// A [`BasicGridView`] entry - a label with background.
+#[derive(Clone, CloneRef, Debug)]
+pub struct Entry {
+    frp:  EntryFrp<Self>,
+    data: Rc<EntryData>,
+}
+
+impl crate::Entry for Entry {
```

Comment:
why do you need to do `crate::Entry` in one place but not in the other place (both are in the same line).

Reply:
Because those are different entities. One is `crate::Entry` and the second is `basic::Entry`. One is a general "Entry" trait, the second is a concrete implementation.